backend/app/utils/real_data_fetchers.py

```
@@ -149,78 +138,49 @@ def _fetch_data(self, symbol: str, days: int) -> pd.DataFrame:
+                df['timestamp'] = pd.to_datetime(df['timestamp'])
+                df['symbol'] = symbol
+                df = df.sort_values('timestamp')
             return df
 
         except Exception as e:
-            logger.error(f"Error fetching data from CoinGecko: {e}")
+            logger.error(f"Error fetching data from CoinMarketCap: {e}")
 
             # Raise an error if API fails
-            raise ValueError(f"Failed to fetch data from CoinGecko for {symbol}: {e}")
+            raise ValueError(f"Failed to fetch data from CoinMarketCap for {symbol}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)